feature-libs/organization/account-summary/components/details/document/account-summary-document.component.ts

```diff
@@ -0,0 +1,145 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import {
+  LanguageService,
+  SortModel,
+  TranslationService,
+} from '@spartacus/core';
+import {
+  AccountSummaryDocumentType,
+  AccountSummaryFacade,
+  AccountSummaryList,
+  DocumentFields,
+  DocumentQueryParams,
+  DocumentStatus,
+  FilterByOptions,
+} from '@spartacus/organization/account-summary/root';
+import { FileDownloadService, ICON_TYPE } from '@spartacus/storefront';
+import { BehaviorSubject, combineLatest, Observable, Subscription } from 'rxjs';
+import { switchMap, take, tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-account-summary-document',
+  templateUrl: './account-summary-document.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AccountSummaryDocumentComponent implements OnInit, OnDestroy {
+  /* For Enum use in HTML */
+  ICON_TYPE = ICON_TYPE;
+
+  documentTypeOptions: AccountSummaryDocumentType[];
+  sortOptions: SortModel[];
+  selectedLanguage: string;
+
+  // Contains the initial query parameters and will be updated with current state of filters
+  _queryParams: DocumentQueryParams = {
+    status: DocumentStatus.OPEN,
+    filterByKey: FilterByOptions.DOCUMENT_NUMBER,
+    page: 0,
+    pageSize: 10,
+    fields: DocumentFields.FULL,
+  };
+  // Used to fire event every time query params are changed
+  queryParams$ = new BehaviorSubject<DocumentQueryParams>(this._queryParams);
+  // Used by template to subscribe to data from documents api
+  accountSummary$: Observable<AccountSummaryList> = this.queryParams$.pipe(
+    switchMap((param) => this.accountSummaryFacade.getDocumentList(param)),
+    tap((accountSummaryList: AccountSummaryList) => {
+      if (accountSummaryList.orgDocumentTypes) {
+        this.documentTypeOptions = accountSummaryList.orgDocumentTypes;
+      }
+
+      if (accountSummaryList.sorts) {
+        this.addNamesToSortModel(accountSummaryList.sorts);
+      }
+    })
+  );
+
+  private subscription = new Subscription();
+
+  constructor(
+    protected accountSummaryFacade: AccountSummaryFacade,
+    protected translation: TranslationService,
+    protected downloadService: FileDownloadService,
+    protected languageService: LanguageService
+  ) {}
+
+  ngOnInit() {
+    this.subscription.add(
+      this.languageService.getActive().subscribe((activeLanguage) => {
+        const reFetchData = !!this.selectedLanguage;
+        this.selectedLanguage = activeLanguage;
+        if (reFetchData) {
```

Comment:
The logic is skip the first `active language`, right? If yes, can use `skip(1)`.

Reply:
That is correct. However, at the moment we can't do that because the selectedLanguage will not have been assigned and is used in the template to format currencies. However backend was recently updated to return pre-formatted string for currencies, so this logic may no longer be needed. Will double check.

Reply:
selectedLanguage was no longer required. I removed it and used skip(1)